resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/DepsTestsSuite.scala

```diff
@@ -125,16 +125,16 @@ private[spark] trait DepsTestsSuite { k8sSuite: KubernetesSuite =>
       .endSpec()
       .build()
 
-    kubernetesTestComponents
+    Eventually.eventually(TIMEOUT, INTERVAL) (kubernetesTestComponents
       .kubernetesClient
       .services()
-      .create(cephService)
+      .create(cephService))
 
-    kubernetesTestComponents
+    Eventually.eventually(TIMEOUT, INTERVAL) (kubernetesTestComponents
```

Comment:
instead of watching for deletion make sure we are able to create the resources if some previous is still being deleted.

Reply:
Maybe make that as a comment since most people won't go back to the PR to understand what is going.

Reply:
I added a comment above of each statement to clarify this.